lib/actions-oclif/device/move.ts

```diff
@@ -82,47 +83,51 @@ export default class DeviceMoveCmd extends Command {
 		options.application = options.application || options.app;
 		delete options.app;
 
-		const device: ExtendedDevice = await balena.models.device.get(
-			params.uuid,
-			expandForAppName,
-		);
-
-		const belongsToApplication = device.belongs_to__application as Application[];
-		device.application_name = belongsToApplication?.[0]
-			? belongsToApplication[0].app_name
-			: 'N/a';
-
-		// Get destination application
 		let application;
-		if (options.application) {
-			application = options.application;
-		} else {
-			const [deviceDeviceType, deviceTypes] = await Promise.all([
-				balena.models.device.getManifestBySlug(device.device_type),
-				balena.models.config.getDeviceTypes(),
-			]);
-
-			const compatibleDeviceTypes = deviceTypes.filter(
-				(dt) =>
-					balena.models.os.isArchitectureCompatibleWith(
-						deviceDeviceType.arch,
-						dt.arch,
-					) &&
-					!!dt.isDependent === !!deviceDeviceType.isDependent &&
-					dt.state !== 'DISCONTINUED',
+		for (const uuid of params.uuid.split(',')) {
+			const device: ExtendedDevice = await balena.models.device.get(
+				uuid,
+				expandForAppName,
 			);
 
-			const patterns = await import('../../utils/patterns');
-			application = await patterns.selectApplication(
-				(app: Application) =>
-					compatibleDeviceTypes.some((dt) => dt.slug === app.device_type) &&
-					// @ts-ignore using the extended device object prop
-					device.application_name !== app.app_name,
-			);
+			const belongsToApplication = device.belongs_to__application as Application[];
+			device.application_name = belongsToApplication?.[0]
+				? belongsToApplication[0].app_name
+				: 'N/a';
+
+			// Get destination application
+			if (!application) {
```

Comment:
My other comment suggests a different overall structure. If that structure was not feasible, then I'm thinking it would easier to read the code if this `if (!application) {...}` block was move outside the `for` loop, perhaps in a function.